pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -538,7 +538,7 @@ func xfrmKeyEqual(s1, s2 *netlink.XfrmState) bool {
 }
 
 func ipSecReplaceStateIn(log *slog.Logger, params *IPSecParameters) (uint8, error) {
-	key := getNodeIPsecKey(*params.SourceTunnelIP, *params.DestTunnelIP, params.LocalBootID, params.RemoteBootID)
+	key := getNodeIPsecKey(*params.SourceTunnelIP, *params.DestTunnelIP, params.RemoteBootID, params.LocalBootID)
```

Comment:
non-blocking-comment/re:internal discussion: the key derivation does seem easy to break. We probably should be able to catch regressions via unit tests and not relying on various e2e.

Reply:
Yup, that might be doable. I've opened https://github.com/cilium/cilium/issues/35843 for it.